frontend/src/components/mine/MineDashboard.js

```diff
@@ -141,8 +161,74 @@ export class MineDashboard extends Component {
     if (this.props.match.params.id !== nextProps.match.params.id) {
       this.loadMineData(id);
     }
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged && !this.state.complianceInfoLoading) {
+      const correctParams = nextProps.location.search
+        ? nextProps.location.search
+        : queryString.stringify(initialSearchValues);
+      this.renderDataFromURL(correctParams);
+    }
   }
 
+  componentWillUnmount() {
+    this.setState({ complianceFilterParams: initialSearchValues });
+  }
+
+  format = (param) => (param ? param.split(",").filter((x) => x) : []);
+
+  renderDataFromURL = (params) => {
+    const { open_orders } = this.props.mineComplianceInfo;
+    const { violation, overdue, ...remainingParams } = queryString.parse(params);
+    const formattedParams = {
+      violation: this.format(violation),
+      overdue: returnBoolean(overdue),
+      ...remainingParams,
+    };
+    const filteredOrders =
+      open_orders.length > 0 &&
+      open_orders.filter((order) => this.handleFiltering(order, formattedParams));
+
+    this.setState({
+      filteredOrders,
+      complianceFilterParams: formattedParams,
+    });
+  };
+
+  handleFiltering = (order, params) => {
+    const overdue = params.overdue === null ? true : order.overdue === params.overdue;
+    const inspector = params.inspector ? order.inspector.includes(params.inspector) : true;
+    const date = params.due_date ? order.due_date.includes(params.due_date) : true;
+    const orderNo = params.order_no ? order.order_no.includes(params.order_no) : true;
+    const reportNoString = order.report_no.toString();
+    const reportNo = params.report_no ? reportNoString.includes(params.report_no) : true;
+    const violation =
+      params.violation.length !== 0 ? params.violation.includes(order.violation) : true;
+    return overdue && inspector && date && orderNo && reportNo && violation;
+  };
+
+  handleComplianceFilter = (values) => {
+    if (isEmpty(values)) {
```

Comment:
How about we refactor this to isolate the difference?
```javascript
const { violation, ...rest } = values;
const urlifiedViolation = violation ? { violation: violation.join(",") } : {};
const params = { ...violation, ...rest };

this.props.history.push(
  router.MINE_SUMMARY.dynamicRoute(
    this.props.match.params.id,
    this.props.match.params.activeTab,
    params
  )
);
```

Reply:
yeah this works! will update it

Reply:
@NWCalvank So, this requires a bit of a bigger change, due to the way `componentWillReceiveProps` is currently set up.
Is this more of a suggestion rather than a must do?